ml/src/test/scala/ai/h2o/sparkling/ml/metrics/BinomialMetricsTestSuite.scala

```diff
@@ -0,0 +1,303 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.metrics
+
+import ai.h2o.sparkling.ml.algos._
+import ai.h2o.sparkling.ml.models.{H2OGBMMOJOModel, H2OGLMMOJOModel, H2OMOJOModel}
+import ai.h2o.sparkling.{SharedH2OTestContext, TestUtils}
+import org.apache.spark.sql.functions.{rand, col}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.types._
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FunSuite, Matchers}
+
+@RunWith(classOf[JUnitRunner])
+class BinomialMetricsTestSuite extends FunSuite with Matchers with SharedH2OTestContext {
+
+  override def createSparkSession(): SparkSession = sparkSession("local[*]")
+
+  import spark.implicits._
+
+  private lazy val dataset = spark.read
+    .option("header", "true")
+    .option("inferSchema", "true")
+    .csv(TestUtils.locate("smalldata/prostate/prostate.csv"))
+    .withColumn("CAPSULE", 'CAPSULE.cast(StringType))
+    .withColumn("RACE", 'RACE.cast(StringType))
+    .withColumn("DCAPS", 'DCAPS.cast(StringType))
+    .withColumn("WEIGHT", rand(42))
+    .repartition(20)
+
+  private lazy val Array(trainingDataset, validationDataset) = dataset.randomSplit(Array(0.8, 0.2), 1234L)
+
+  private def assertMetrics[T](model: H2OMOJOModel): Unit = {
+    assertMetrics[T](model.getTrainingMetricsObject(), model.getTrainingMetrics())
+    assertMetrics[T](model.getValidationMetricsObject(), model.getValidationMetrics())
+    assert(model.getCrossValidationMetricsObject() == null)
+    assert(model.getCrossValidationMetrics() == Map())
+  }
+
+  private def assertMetrics[T](metricsObject: H2OMetrics, metrics: Map[String, Double]): Unit = {
+    metricsObject.isInstanceOf[T] should be(true)
+    MetricsAssertions.assertMetricsObjectAgainstMetricsMap(metricsObject, metrics)
+    val binomialObject = metricsObject.asInstanceOf[H2OBinomialMetrics]
+    binomialObject.getConfusionMatrix().count() > 0
+    binomialObject.getConfusionMatrix().columns.length > 0
+    binomialObject.getGainsLiftTable().count() > 0
+    binomialObject.getGainsLiftTable().columns.length > 0
+    binomialObject.getMaxCriteriaAndMetricScores().count() > 0
+    binomialObject.getMaxCriteriaAndMetricScores().columns.length > 0
+    binomialObject.getThresholdsAndMetricScores().count() > 0
+    binomialObject.getThresholdsAndMetricScores().columns.length > 0
```

Comment:
won't all those lines ending with `> 0` be just unused boolean statements? maybe `binomialObject.getConfusionMatrix() should not be empty` etc.?

Reply:
Good catch, thank you!